generator/sbpg/targets/resources/sbp_java.java.j2

```diff
@@ -52,7 +52,7 @@ public class (((m.identifier | classnameify))) extends SBPMessage {
     public (((m.identifier | classnameify))) () { super(TYPE); }
     public (((m.identifier | classnameify))) (SBPMessage msg) throws SBPBinaryException {
         super(msg);
-        assert msg.type != TYPE;
+        assert msg.type == TYPE;
```

Comment:
🤔 how did this ever work? Do asserts not fire in certain cases?

Reply:
no ;-)

I assume the existing usage/examples doesnt go through this constructor, as it uses SBPHandler.